packages/itwinui-react/src/core/Alert/Alert.tsx

```diff
@@ -6,26 +7,135 @@ import cx from 'classnames';
 import * as React from 'react';
 import '@itwin/itwinui-css/css/alert.css';
 import { useTheme, StatusIconMap, SvgCloseSmall } from '../utils/index.js';
-import type { CommonProps } from '../utils/props.js';
+import type {
+  CommonProps,
+  PolymorphicComponentProps,
+  PolymorphicForwardRefComponent,
+} from '../utils/props.js';
 
-export type AlertProps = {
+// ----------------------------------------------------------------------------
+// Alert.Icon component
+
+type AlertIconOwnProps = {
   /**
-   * Type of the alert.
-   * @default 'informational'
+   * Type of the alert icon.
    */
   type?: 'positive' | 'warning' | 'negative' | 'informational';
+};
+
+const AlertIcon = React.forwardRef((props, ref) => {
+  const { as: Element = 'svg', children, className, type, ...rest } = props;
+
+  if (type) {
+    const StatusIcon = StatusIconMap[type];
+    return (
+      <StatusIcon
+        className={cx('iui-alert-icon', className)}
+        ref={ref}
+        {...rest}
+      />
+    );
+  }
+
+  return (
+    <Element className={cx('iui-alert-icon', className)} ref={ref} {...rest}>
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'svg', AlertIconOwnProps>;
+
+// ----------------------------------------------------------------------------
+// Alert.Message component
+
+type AlertMessageOwnProps = {};
+
+const AlertMessage = React.forwardRef((props, ref) => {
+  const { as: Element = 'span', children, className, ...rest } = props;
+
+  return (
+    <Element className={cx('iui-alert-message', className)} ref={ref} {...rest}>
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'span', AlertMessageOwnProps>;
+
+// ----------------------------------------------------------------------------
+// Alert.ClickableText component
+
+type AlertClickableTextOwnProps = {};
+
+const AlertClickableText = React.forwardRef((props, ref) => {
+  const { as: Element = 'a', children, className, ...rest } = props;
+
+  return (
+    <Element className={cx('iui-alert-link', className)} ref={ref} {...rest}>
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'a', AlertClickableTextOwnProps>;
+
+// ----------------------------------------------------------------------------
+// Alert.CloseButton component
+
+type AlertCloseButtonOwnProps = {
   /**
-   * Text for the link you want to provide.
-   */
-  clickableText?: React.ReactNode;
-  /**
-   * Props for the clickable text. Used for providing `href` and other attributes.
+   * Action handler for close.
    */
-  clickableTextProps?: React.ComponentPropsWithRef<'a'>;
+  onClose: () => void;
```

Comment:
no need for `onClose`; `onClick` should be enough

Reply:
Removed